File: Herança - Subclasses de Personagem/aula4_heranca_JAVA/Item.java
package aula4_heranca_JAVA;

public class Item {

    private String nome;
    private int bonus;

public Item(String nome, int bonus){
    setNome(nome);
    setBonus(bonus);
}
public String getNome(){
    return nome;
}
public void setNome(String nome){
if (nome == null || nome.isEmpty()){
    System.err.println("Nome inválido");
}else{
    this.nome = nome;
}
}
public int getBonus(){
    return bonus;
}
public void setBonus(int b){
    if (b >= 0){
         this.bonus = b;
    } else {
       System.err.println("Bonus inválido");
    }
}
public String descricao() {
    return nome + " (+" + bonus + ")";
}
}

File: Polimorfismo - Batalha Final/python/item.py
class Item:
    def __init__(self, nome, bonus):
        self.set_nome(nome)
        self.set_bonus(bonus)

    def get_nome(self):
        return self.nome

    def set_nome(self, nome):
        if not nome:
            print("Nome inválido")
        else:
            self.nome = nome

    def get_bonus(self):
        return self.bonus

    def set_bonus(self, bonus):
        if bonus >= 0:
            self.bonus = bonus
        else:
            print("Bonus inválido")

    def descricao(self):
        return f"{self.nome} (+{self.bonus})"